bcipy/helpers/triggers.py

```diff
@@ -638,3 +639,185 @@ def read_triggers(triggers_file: TextIO) -> List[Tuple[str, str, float]]:
+
+    def close(self):
+        self.write()
+        self.file.close()
+
+    def write(self):
+        """
+        Writes current Triggers in self.triggers[] to .txt file in self.file_name.
+        File writes in the format "label, targetness, time".
+
+        Parameters
+        ----------
+        None
+
+        Returns
+        -------
+            None
+        """
+
+        for trigger in self.triggers:
+            self.file.write(f'{trigger.label} {trigger.type.value} {trigger.time}\n')
+
+        self.triggers = []
+
+    @staticmethod
+    def read_text_file(path: str) -> List[str]:
+        if not path.endswith('.txt') or not os.path.exists(path):
+            raise FileNotFoundError(f"Valid triggers .txt file not found at [{path}]."
+                                    "\nPlease rerun program.")
+
+        triggers_list = []
+        with open(path) as raw_txt:
+            for line in raw_txt:
+                line_split = line.split()
+                triggers_list.append(line_split)
+        return triggers_list
+
+    @staticmethod
+    def load(path: str,
+             offset: Optional[float] = None,
+             exclusion: Optional[List[TriggerType]] = None) -> List[Trigger]:
+        """
+        Loads a list of triggers from a .txt of triggers.
+
+        Exclusion based on type only (ex. exclusion=[TriggerType.Fixation])
+
+        1. Checks if .txt file exists at path
+        2. Loads the triggers data as a list of lists
+        3. If offset provided, adds it to the time as float
+        4. If exclusion provided, filters those triggers
+        5. Casts all loaded and modified triggers to Trigger
+        6. Returns as a List[Triggers]
+
+        Parameters
+        ----------
+        path (str): name or file path of .txt trigger file to be loaded.
+            Input string must include file extension (.txt).
+        offset (Optional float): if desired, time offset for all loaded triggers,
+            positive number for adding time, negative number for subtracting time.
+        exclusion (Optional List[TriggerType]): if desired, list of TriggerTypes
+            to be removed from the loaded trigger list.
+
+        Returns
+        -------
+            List of Triggers from loaded .txt file with desired modifications
+        """
+
+        # Checking for file with given path, with or without .txt
+        triggers_list = TriggerHandler.read_text_file(path)
+
+        if offset:
+            # If there is exclusion but no offset,
+            # Program would read exclusion as offset and throw error
+            # This moves "offset" into its proper parameter of exclusion
+            if isinstance(offset, list) and exclusion is None:
+                exclusion = offset
+            else:
+                for item in triggers_list:
+                    time_float = float(item[2]) + offset
+                    item[2] = str(time_float)
+
+        if exclusion:
+            for type in exclusion:
```

Comment:
This variable could be renamed `trigger_type` so we're not unintentionally redefining the built-in `type` function.

Reply:
I don't know if it's too unwieldy or makes enough sense, but maybe something like `category` would also work? I'm not sure  if, in this context of possible confusion with built-in `type`,  `trigger.category` would be distinct enough from `trigger.type`. I think both possibilities would require a bit of refactoring with the `Enum` and such as well, though.

Reply:
I don't have a strong preference on the variable name. Maybe @sci-tab has an opinion here?